Two more split commands

Nice extension which works well for my purpose.

One feature which will save me even more time, is to SplitBeforeDelimiter or SplitAfterDelimiter. It means that the delimiter will stay at Begin or End of Line, respectively.
My example:

When I wrap a list on multiple lines, I want to keep the comma. I know I can add it back with a search& replace, but it will be better if it is not eaten at all.

First example, constants set to reformat:
--------------------------------

```{plenty_of_long, but_variable_in_length, name_for_a_useful_project}```
after **SplitAfterDelimiter(",")** would give
```
{plenty_of_long,
 but_variable_in_length,
 name_for_a_useful_project}
```

Second example, splitting an options list for readability will keep all the details:
--------------------------------

```-e one=two -e three=four -e do_it=again -v -t 30```
after **SplitBeforeDelimiter(" -")** would give

```
 -e one=two
 -e three=four
 -e do_it=again
 -v
 -t 30
```

Thanks for reading

Hello,

I tried to prepare a pull-request, but got an error:

```
[22:15:10] Starting compilation in watch mode...

src/Transforms.ts:4:40 - error TS2307: Cannot find module 'vscode-extension-common' or its corresponding type declarations.

4 import {Lines,Modify,Region,View} from 'vscode-extension-common'
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~
``` 

Is there a trick to set this up, or some hidden prerequisites ?

Thanks for your help

--- a/package.json
+++ b/package.json
@@ -42,6 +42,8 @@
         "onCommand:dakara-transformer.sortByLineLength",
         "onCommand:dakara-transformer.sortLines",
         "onCommand:dakara-transformer.splitLines",
+        "onCommand:dakara-transformer.splitLinesBeforeDelimiter",
+        "onCommand:dakara-transformer.splitLinesAfterDelimiter",
         "onCommand:dakara-transformer.trimLines",
         "onCommand:dakara-transformer.uniqueLines",
         "onCommand:dakara-transformer.uniqueLinesNewDocument"
@@ -192,6 +194,16 @@
             {
                 "command": "dakara-transformer.splitLines",
                 "title": "Split Lines...",
+                "category": "Transform"
+            },
+            {
+                "command": "dakara-transformer.splitLinesBeforeDelimiter",
+                "title": "Split Lines Before...",
+                "category": "Transform"
+            },
+            {
+                "command": "dakara-transformer.splitLinesAfterDelimiter",
+                "title": "Split Lines After...",
                 "category": "Transform"
             }
         ],

--- a/src/Transforms.ts
+++ b/src/Transforms.ts
@@ -397,6 +397,18 @@ export async function splitLines(textEditor: vscode.TextEditor) {
     Modify.replaceUsingTransform(textEditor, textEditor.selections, text => text.split(userInput).join(splitChar))
 }
 
+export async function splitLinesBeforeDelimiter(textEditor: vscode.TextEditor) {
+    const userInput = await vscode.window.showInputBox({prompt:'Specify Delimiter', value: ''});
+    const splitChar = Lines.lineEndChars(textEditor) + userInput
+    Modify.replaceUsingTransform(textEditor, textEditor.selections, text => text.split(userInput).join(splitChar))
+}
+
+export async function splitLinesAfterDelimiter(textEditor: vscode.TextEditor) {
+    const userInput = await vscode.window.showInputBox({prompt:'Specify Delimiter', value: ''});
+    const splitChar =  userInput + Lines.lineEndChars(textEditor)
+    Modify.replaceUsingTransform(textEditor, textEditor.selections, text => text.split(userInput).join(splitChar))
+}
+
 export function escapes(textEditor: vscode.TextEditor) {
     const selections = textEditor.selections
 

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -208,6 +208,16 @@ export function activate(context: vscode.ExtensionContext) {
         transforms.splitLines(textEditor);
     });
 
+    Application.registerCommand(context, 'dakara-transformer.splitLinesBeforeDelimiter', () => {
+        const textEditor = vscode.window.activeTextEditor;
+        transforms.splitLinesBeforeDelimiter(textEditor);
+    });
+
+    Application.registerCommand(context, 'dakara-transformer.splitLinesAfterDelimiter', () => {
+        const textEditor = vscode.window.activeTextEditor;
+        transforms.splitLinesAfterDelimiter(textEditor);
+    });
+
 
 }
 

